core/amber/src/main/scala/edu/uci/ics/texera/workflow/WorkflowCompiler.scala

```diff
@@ -211,9 +205,7 @@ class WorkflowCompiler(
           if (sink.getChartType.contains(VisualizationConstants.HTML_VIZ)) OpResultStorage.MEMORY
           else OpResultStorage.defaultStorageMode
         }
-        if (reuseStorageSet.contains(storageKey) && storage.contains(storageKey)) {
-          sink.setStorage(storage.get(storageKey))
-        } else {
+        if (!reuseStorageSet.contains(storageKey) || !storage.contains(storageKey)) {
```

Comment:
I think you are removing part of the cache logic here.

Reply:
we don't set storage anymore. we only set upstream op id.